### PR TITLE
Add initial version of http transport

### PR DESCRIPTION
**Pull request by gotha**

### CODE REVIEW

File: transport/http/lifecycle.go [hunk 1]
```diff
@@ -0,0 +1,20 @@
+package http
+
+import "context"
+
+type ServerLifecycle struct {
+	srv *Server
+}
+
+func NewServerLifecycle(srv *Server) *ServerLifecycle {
```
Comment (atanasdinov) [inline transport/http/lifecycle.go:9]: Don't we want to return the `Lifecycle` interface from the `app` package?

**Review by atanasdinov — CHANGES_REQUESTED** [review]
Up for discussion:

- The decoder functions are looking good however shouldn't we have one constructing a "hybrid" request from both query params and body?
- The `apperror` package is too complex with all the wrapping and unwrapping functions. My suggestion is simplifying those to `New(errType ErrorType, message string, args... interface{})` and `Wrap(errType ErrorType, err error, message string, args... interface{})`. Then we have the option of using formatted messages without additional functions and possibly removing the `github.com/pkg/errors` package altogether. To my understanding `AppError` will be constructed as a final error from a given action e.g. concluding a HTTP response. We wouldn't be flooding our services with `AppError` wrapping on each possible error.

File: transport/http/server.go [hunk 2]
```diff
@@ -0,0 +1,128 @@
...
+func (s *Server) Start() error {
+	return s.srv.ListenAndServe()
+}
+
+// Close - stop server
+func (s *Server) Close() error {
+	return s.srv.Close()
+}
+
+func statusCodeFromError(err error) int {
```
Comment (gotha) [inline transport/http/server.go:117]: maybe this should be an option that can be overridden

File: transport/http/decoder.go [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+			return nil, fmt.Errorf("error decoding request: %w", err)
+		}
+
+		return req, nil
+	}
+}
+
+func DecodeRequestFromJSONBody(req interface{}) httptransport.DecodeRequestFunc {
+	return func(_ context.Context, httpReq *http.Request) (interface{}, error) {
+		body, err := ioutil.ReadAll(httpReq.Body)
```
[the remark this answers is not in the document]
Comment (gotha) [reply transport/http/decoder.go:44]: I am receiving the request, not sending it. There is no need to close the body. 
```
        // Body is the request's body.
	//
	// For client requests, a nil body means the request has no
	// body, such as a GET request. The HTTP Client's Transport
	// is responsible for calling the Close method.
	//
	// For server requests, the Request Body is always non-nil
	// but will return EOF immediately when no body is present.
	// The Server will close the request body. The ServeHTTP
	// Handler does not need to.
	//
	// Body must allow Read to be called concurrently with Close.
	// In particular, calling Close should unblock a Read waiting
	// for input.
	Body io.ReadCloser
```

**Review by atanasdinov — APPROVED** [review]

**Pull request merged**